tests/test_connector.py

```diff
@@ -1216,6 +1221,67 @@ def capture_future(*args, features_future=None, **kwargs):
             )
         )
 
+        # test for correct generation of warnings if set_alpn_proto is
```

Comment:
This should be a separate test case. If needed, you can refactor the context factory out of the connect method into a method of the object. That might be saner anyways. Allows to refactor the test for that context factory wrapper out into separate tests, too. Then we only need to test against correct use of functools partial, if that’s even needed.

Reply:
Yes, factoring out the mock setup sounds right. I did it this way, because I considered avoiding the code duplication to be more important than having separate test cases, don't ask why I did not think of factoring the setup out.